src/cls/rbd/cls_rbd_types.h

```diff
@@ -293,10 +291,29 @@ struct GroupSnapshotNamespace {
     }
     return false;
   }
-
 };
 
-std::ostream& operator<<(std::ostream& os, const GroupSnapshotNamespace& ns);
+struct TrashSnapshotNamespace {
+  static const SnapshotNamespaceType SNAPSHOT_NAMESPACE_TYPE =
+    SNAPSHOT_NAMESPACE_TYPE_TRASH;
+
+  std::string original_name;
```

Comment:
@dillaman Just wondering how is original_name going to be used? Just for diagnostic? Don't we want to store original_namespace (type)?

Reply:
I still have more work to do in future PR(s) -- I figured at some point it might be nice to be able to retrieve the original snapshot name via the API/CLI (since we rename them to ensure uniqueness).

I was on the fence about adding the namespace because I couldn't really immediately figure out how it could be useful since if it's in the trash the and it's a GROUP (or some future snapshot type), it's already been unlinked. Perhaps just the namespace type enum would be useful. Thoughts?

Reply:
Storing at least original namespace type enum looks useful to me. I am not sure about storing original namespace, if it has much sense. On the other hand, it does not look like much overhead, thus why not, just in case? I suppose it would be harder to add it later if we needed it?

Reply:
OK -- I'll append a commit to store the enum (since storing the whole namespace variant recursively within a namespace is non-trivial and if needed could be added in the future).

Reply:
Ah, fair enough.